frontend/app_flowy/packages/appflowy_editor/lib/src/service/internal_key_event_handlers/checkbox_event_handler.dart

```diff
@@ -0,0 +1,22 @@
+import 'package:appflowy_editor/src/service/shortcut_event/shortcut_event_handler.dart';
+import 'package:appflowy_editor/src/core/document/attributes.dart';
+import 'package:appflowy_editor/src/core/document/node.dart';
+import 'package:flutter/material.dart';
+
+ShortcutEventHandler toggleCheckbox = (editorState, event) {
+  final selection = editorState.service.selectionService.currentSelection.value;
+  final nodes = editorState.service.selectionService.currentSelectedNodes;
+  final textNodes = nodes.whereType<TextNode>().toList(growable: false);
+  if (selection == null || textNodes.isEmpty) {
+    return KeyEventResult.ignored;
+  }
+  for (Node node in textNodes) {
```

Comment:
It's not allowed to use `updateAttributes` function directly.

You should use transactions to update the document.

For example
```dart
final transaction = editorState.transaction;
  for (final node in checkboxTextNodes) {
    transaction.updateNode(node, {
      BuiltInAttributeKey.checkbox:
          !node.attributes[BuiltInAttributeKey.checkbox]
    });
  }
  editorState.apply(transaction);
```